checker/src/main/java/org/checkerframework/checker/calledmethods/CalledMethodsAnalysis.java

```diff
@@ -20,21 +22,31 @@
   protected CalledMethodsAnalysis(
       BaseTypeChecker checker, CalledMethodsAnnotatedTypeFactory factory) {
     super(checker, factory);
+    // Use the Nullness Checker.
+    ignoredExceptionTypes.add("java.lang.NullPointerException");
+    // Ignore run-time errors, which cannot be predicted statically. Doing so is unsound
+    // in the sense that they could still occur - e.g., the program could run out of memory -
+    // but if they did, the checker's results would be useless anyway.
+    ignoredExceptionTypes.add("java.lang.Error");
+    ignoredExceptionTypes.add("java.lang.RuntimeException");
```

Comment:
Why are we adding these exception types in this PR?  Do the tests not pass without it?

Reply:
The tests do not pass without it.

In particular, we get these three false positives:
```
    3 unexpected diagnostics were found
    CheckFields.java:68: error: (destructor.exceptional.postcondition)
    CreatesMustCallForOverride2.java:38: error: (destructor.exceptional.postcondition)
    CreatesMustCallForOverride2.java:56: error: (destructor.exceptional.postcondition)
```

For reference, here's the relevant lines of `CheckFields`:
```
    @EnsuresCalledMethods(
        value = {"this.finalOwningFoo", "this.owningFoo"},
        methods = {"a"})
    void b() {
      this.finalOwningFoo.a();
      this.finalOwningFoo.c();
      this.owningFoo.a();
    }
  }
```

Reply:
After this lands we may want to think about another manual update like #4892

Reply:
I'm not sure that's really necessary - that run-time errors don't invalidate CalledMethods types follows from the idea that annotations (and therefore types) specify *normal behavior*, and a run-time exception is definitely not normal behavior.